src/Repositories/TranslationRepository.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+namespace Brackets\AdminTranslations\Repositories;
+
+use Brackets\AdminTranslations\Translation;
+
+class TranslationRepository
+{
+    public function createOrUpdate($namespace, $group, $key, $language, $text)
+    {
+        /** @var Translation $translation */
+        $translation = Translation::withTrashed()
```

Comment:
In my opinion only this should be in repository, to return collection of translations. Other stuff can be in service.